dbms/src/TestUtils/mockExecutor.cpp

```diff
@@ -0,0 +1,256 @@
+        ASTPtr locale_node;
+        auto order_by_item = std::make_shared<ASTOrderByElement>(direction, direction, false, locale_node);
+        order_by_item->children.push_back(std::make_shared<ASTIdentifier>(item.first));
+        vec.push_back(order_by_item);
+    }
+    auto exp_list = std::make_shared<ASTExpressionList>();
+    for (const auto & ast_ptr : vec)
+        exp_list->children.push_back(ast_ptr);
+    return exp_list;
+}
+
+void DAGRequestBuilder::initDAGRequest(tipb::DAGRequest & dag_request)
+{
+    dag_request.set_time_zone_name(properties.tz_name);
+    dag_request.set_time_zone_offset(properties.tz_offset);
+    dag_request.set_flags(dag_request.flags() | (1u << 1u /* TRUNCATE_AS_WARNING */) | (1u << 6u /* OVERFLOW_AS_WARNING */));
+
+    if (properties.encode_type == "chunk")
+        dag_request.set_encode_type(tipb::EncodeType::TypeChunk);
+    else if (properties.encode_type == "chblock")
+        dag_request.set_encode_type(tipb::EncodeType::TypeCHBlock);
+    else
+        dag_request.set_encode_type(tipb::EncodeType::TypeDefault);
+
+    for (size_t i = 0; i < root->output_schema.size(); ++i)
+        dag_request.add_output_offsets(i);
+}
+
+std::shared_ptr<tipb::DAGRequest> DAGRequestBuilder::build(Context & context)
+{
+    MPPInfo mpp_info(properties.start_ts, -1, -1, {}, {});
+    std::shared_ptr<tipb::DAGRequest> dag_request_ptr = std::make_shared<tipb::DAGRequest>();
+    tipb::DAGRequest & dag_request = *dag_request_ptr;
+    initDAGRequest(dag_request);
+    root->toTiPBExecutor(dag_request.mutable_root_executor(), properties.collator, mpp_info, context);
+
+    return dag_request_ptr;
+}
+
+DAGRequestBuilder & DAGRequestBuilder::mockTable(const String & db, const String & table, const MockColumnInfos & columns)
+{
+    assert(!columns.empty());
+    TableInfo table_info;
+    table_info.name = db + "." + table;
+    for (const auto & column : columns)
+    {
+        TiDB::ColumnInfo ret;
+        ret.tp = column.second;
+        ret.name = column.first;
+        table_info.columns.push_back(std::move(ret));
+    }
+    String empty_alias;
+    root = compileTableScan(getExecutorIndex(), table_info, empty_alias, false);
+    return *this;
+}
+
+DAGRequestBuilder & DAGRequestBuilder::mockTable(const MockTableName & name, const std::vector<std::pair<String, TiDB::TP>> & columns)
+{
+    return mockTable(name.first, name.second, columns);
+}
+
+DAGRequestBuilder & DAGRequestBuilder::mockTable(const MockTableName & name, const MockColumnInfoList & columns)
+{
+    assert(columns.size() != 0);
+    TableInfo table_info;
+    table_info.name = name.first + "." + name.second;
+    for (const auto & column : columns)
+    {
+        TiDB::ColumnInfo ret;
+        ret.tp = column.second;
+        ret.name = column.first;
+        table_info.columns.push_back(std::move(ret));
+    }
+    String empty_alias;
+    root = compileTableScan(getExecutorIndex(), table_info, empty_alias, false);
+    return *this;
+}
```

Comment:
Can be replaced by `return mockTable(name.first, name.second, columns);` ?

Reply:
Fixed it.